### PR TITLE
Fix SpinDensityNew HDF5 writing

### PR DESCRIPTION
**Pull request by PDoakORNL**
## Proposed changes

Thanks to @jtkrogel for updating Nexus and his observations on the source and output this was pretty easy to fix.

HDF5 writing from EstimatorManagerNew is essentially unchanged from legacy until this PR. It assumes anything you'd write from an estimator is in the Data owned by the Manager in the various Cache blocks.  It isn't for the OperatorEstimators, this is now handled.

Several bugs involving zeroing the estimator data between blocks are fixed.

Regression involving asserts in Walker code necessitates commenting out test_WalkerControlMPI.cpp tests.

## What type(s) of changes does this code introduce?

- Bugfix
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Leconte

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- Yes This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes Documentation has been added (if appropriate)

### CODE REVIEW

File: src/Estimators/EstimatorManagerNew.h [hunk 1]
```diff
@@ -269,7 +270,9 @@ class EstimatorManagerNew
   void makeBlockAverages(unsigned long accept, unsigned long reject);
 
   void reduceOperatorEstimators();
-
+  void writeOperatorEstimators();
```
Comment (ye-luo) [inline src/Estimators/EstimatorManagerNew.h:273]: Need documentation.

File: src/Estimators/OperatorEstBase.h [hunk 2]
```diff
@@ -88,15 +88,24 @@ class OperatorEstBase
...
+   *
+   * Since Operator Estimators own their data they can't just stuff an index in here and call it a day
+   * they need to have the indexes for their h5desc.  This whole hdf5 design could use some help.
    */
-  virtual void registerOperatorEstimator(std::vector<observable_helper*>& h5desc, hid_t gid) const {}
+  virtual void registerOperatorEstimator(hid_t gid) {}
 
   virtual OperatorEstBase* clone() = 0;
 
+  void write();
```
Comment (ye-luo) [inline src/Estimators/OperatorEstBase.h:99]: Need documentation.

**Review by ye-luo — CHANGES_REQUESTED** [review]
better to merge #2866 first. Blocking merge for the moment.

**Comment by PDoakORNL** [comment]
I believe the requested changes are now done

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**